frontend/src/metabase/query_builder/components/ResponsiveParametersList.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useCallback, useState, useMemo } from "react";
+import useIsSmallScreen from "metabase/hooks/use-is-small-screen";
+import Button from "metabase/core/components/Button";
+
+import { Parameter } from "metabase-types/types/Parameter";
+
+import {
+  FilterButton,
+  ParametersListHeader,
+  StyledParametersList,
+  ResponsiveParametersListRoot,
+  ParametersListContainer,
+} from "./ResponsiveParametersList.styled";
+
+interface ResponsiveParametersListProps {
+  parameters: Parameter[];
+  setParameterValue: (parameterId: string, value: string) => void;
+  setParameterIndex: (parameterId: string, parameterIndex: number) => void;
+}
+
+export const ResponsiveParametersList = ({
+  parameters,
+  setParameterValue,
+  setParameterIndex,
+}: ResponsiveParametersListProps) => {
+  const [mobileShowParameterList, setShowMobileParameterList] = useState(false);
+  const isSmallScreen = useIsSmallScreen();
+
+  const handleFilterButtonClick = useCallback(() => {
+    setShowMobileParameterList(!mobileShowParameterList);
+  }, [mobileShowParameterList]);
```

Comment:
Another pro-tip! Use functional setState to eliminate changing this callback every time the state itself changed.

```suggestion
    setShowMobileParameterList(mobileShowParameterList => !mobileShowParameterList);
  }, []);
```